pkg/il/interpreter/extern.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package interpreter
+
+import (
+	"reflect"
+
+	"istio.io/mixer/pkg/il"
+)
+
+// Extern represents an external, native function that is callable from within the interpreter,
+// during program execution.
+type Extern struct {
```

Comment:
Absolutely, and I think that is the right way to go. Initially, I had such a model here, but the glue code looked extremely ugly and I didn't want to spend any time trying to come up with a decent model for this as it is not needed yet. Think of this as a basic/generalized/proof/fallback implementation for an extern infrastructure.